ixwebsocket/IXWebSocketPoll.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ *  IXWebSocketPoll.cpp
+ *  Author: Korchynskyi Dmytro
+ *  Copyright (c) 2017-2018 Machine Zone, Inc. All rights reserved.
+ */
+
+#include "IXWebSocketPoll.h"
+
+namespace ix
+{
+
+    WebSocketPoll::WebSocketPoll(WebSocket* websocket)
+    {
+        bindWebsocket(websocket);
+    }
+
+    WebSocketPoll::~WebSocketPoll()
+    {
+        if (!_messages.empty())
+        {
+            // not handled all messages
+        }
+
+        bindWebsocket(nullptr);
+    }
+
+    void WebSocketPoll::bindWebsocket(WebSocket * websocket)
+    {
+        if (_websocket != websocket)
+        {
+            // unbind old
+            if (_websocket)
+            {
+                _websocket->setOnMessageCallback(nullptr);
+            }
+
+            _websocket = websocket;
+
+            // bind new
```

Comment:
With a shared_ptr we probably don't need this null check right ?

Reply:
Still need because shared_ptr also can be null